config/url.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"net/url"
+	"strings"
+)
+
+func NormalizeURL(address string) (*url.URL, error) {
+	// Trim any leading or trailing whitespace
+	address = strings.TrimSpace(address)
+
+	// Parse the URL
+	u, err := url.Parse(address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Default to https
+	u.Scheme = "https"
+	if u.Scheme != "" {
+		// Force https if http was specified
+		u.Scheme = "https"
+	}
```

Comment:
do we need lines 20-23? After you init it at line 19,  the check at line 20 is true.